opentimelineview/timeline_widget.py

```diff
@@ -338,35 +338,45 @@ def __init__(self, stack, *args, **kwargs):
         self._add_markers()
 
     def _adjust_scene_size(self):
-        # Make the scene large enough so that all items fit
-        all_ranges = [t.range_of_child_at_index(n)
-                      for t in self.stack for n in range(len(t))]
-
-        if all_ranges:
-            start_time = min(c.start_time for c in all_ranges)
-            end_time_exclusive = max(
-                c.end_time_exclusive() for c in all_ranges
-            )
-        else:
-            start_time = otio.opentime.RationalTime()
-            end_time_exclusive = otio.opentime.RationalTime()
+        scene_range = self.composition.trimmed_range()
 
-        start_time = otio.opentime.to_seconds(start_time)
-        duration = otio.opentime.to_seconds(end_time_exclusive)
+        start_time = otio.opentime.to_seconds(scene_range.start_time)
+        duration = otio.opentime.to_seconds(scene_range.end_time_exclusive())
 
-        has_video_tracks = any(
-            t for t in self.stack if t.kind == otio.schema.SequenceKind.Video
-        )
-        has_audio_tracks = any(
-            t for t in self.stack if t.kind == otio.schema.SequenceKind.Audio
-        )
+        if isinstance(self.composition, otio.schema.Stack):
+            # non audio tracks are sorted into one area
+            has_video_tracks = any(
+                t.kind != otio.schema.SequenceKind.Audio
+                for t in self.composition
+            )
+            has_audio_tracks = any(
+                t.kind == otio.schema.SequenceKind.Audio
+                for t in self.composition
+            )
+        elif isinstance(self.composition, otio.schema.SequenceKind):
+            has_video_tracks = (
+                self.composition.kind != otio.schema.SequenceKind.Audio
```

Comment:
Is it required otio.schema.SequenceKind.Video?

Reply:
ah, actually look a bit closer - one is == and the other one is !=.  So this is saying anything that isn't an audio track should be assumed to be a video track, and _only_ things marked as audio tracks should be assumed to be audio tracks.  Have you seen this cause a problem with something?